src/transformStateWithClones.js

```diff
@@ -6,8 +6,39 @@
  *
  * @return {Object[]}
  */
-function transformStateWithClones(state, actions) {
-  // write code here
+function transformStateWithClones(initialState, actions) {
+  let currentState = { ...initialState };
+  const states = [];
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'clear':
+        currentState = {};
+        break;
+
+      case 'addProperties':
+        currentState = {
+          ...currentState,
+          ...action.extraData,
+        };
+        break;
+
+      case 'removeProperties':
+        currentState = { ...currentState };
+
+        for (const key of action.keysToRemove) {
+          delete currentState[key];
+        }
+        break;
+
+      default:
+        throw new Error(`Неизвестный тип действия: ${action.type}`);
+    }
+
+    states.push(currentState);
```

Comment:
The current implementation pushes `currentState` directly to the `states` array. According to the additional prompt rules, you should push a copy of the `currentState` object to the `states` array at the end of each loop cycle, but outside of the `switch` block. This ensures that each state is captured correctly after processing each action.